waiter/src/waiter/scheduler/marathon.clj

```diff
@@ -409,25 +565,36 @@
+  "Returns a new MarathonScheduler with the provided configuration.
+   Validates the configuration against marathon-scheduler-schema and throws if it's not valid."
   [{:keys [home-path-prefix http-options force-kill-after-ms framework-id-ttl mesos-slave-port
-           service-id->service-description-fn slave-directory url is-waiter-app?-fn]}]
+           sync-deployment service-id->service-description-fn slave-directory url
+           is-waiter-app?-fn leader?-fn]}]
   {:pre [(not (str/blank? url))
          (or (nil? slave-directory) (not (str/blank? slave-directory)))
          (or (nil? mesos-slave-port) (utils/pos-int? mesos-slave-port))
          (utils/pos-int? framework-id-ttl)
          (utils/pos-int? (:conn-timeout http-options))
          (utils/pos-int? (:socket-timeout http-options))
-         (not (str/blank? home-path-prefix))]}
+         (not (str/blank? home-path-prefix))
+         (utils/pos-int? (:interval-ms sync-deployment))
+         (utils/pos-int? (:timeout-ms sync-deployment))
+         (<= (:interval-ms sync-deployment) (:timeout-ms sync-deployment))]}
   (when (or (not slave-directory) (not mesos-slave-port))
     (log/info "scheduler mesos-slave-port or slave-directory is missing, log directory and url support will be disabled"))
   (let [http-client (http-utils/http-client-factory http-options)
         marathon-api (marathon/api-factory http-client http-options url)
         mesos-api (mesos/api-factory http-client http-options mesos-slave-port slave-directory)
         service-id->failed-instances-transient-store (atom {})
         service-id->last-force-kill-store (atom {})
-        retrieve-framework-id-fn (memo/ttl #(retrieve-framework-id marathon-api) :ttl/threshold framework-id-ttl)]
-    (->MarathonScheduler marathon-api mesos-api retrieve-framework-id-fn home-path-prefix
-                         service-id->failed-instances-transient-store service-id->last-force-kill-store
-                         service-id->service-description-fn force-kill-after-ms is-waiter-app?-fn)))
+        service-id->out-of-sync-state-store (atom {})
+        retrieve-framework-id-fn (memo/ttl #(retrieve-framework-id marathon-api) :ttl/threshold framework-id-ttl)
+        sync-deployment-maintainer-atom (atom nil)
+        marathon-scheduler (->MarathonScheduler
+                             marathon-api mesos-api retrieve-framework-id-fn home-path-prefix
+                             service-id->failed-instances-transient-store service-id->last-force-kill-store
+                             service-id->out-of-sync-state-store service-id->service-description-fn
+                             force-kill-after-ms is-waiter-app?-fn sync-deployment-maintainer-atom)
+        sync-deployment-maintainer (start-sync-deployment-maintainer leader?-fn marathon-scheduler sync-deployment)]
+    (reset! sync-deployment-maintainer-atom sync-deployment-maintainer)
```

Comment:
We need this to be able to return the `sync-deployment-maintainer` channels to the caller via the `scheduler` instance.